launcher/cli/utils.go

```diff
@@ -44,5 +51,154 @@ func getDirsToMake(storeURL string) []string {
+		return
+	}
+
+	return newNodeosVersionFromString(string(stdOut))
+}
+
+// NewNodeosVersionFromString parsed the received string and return a structured object
+// representing the version information.
+func newNodeosVersionFromString(version string) (out nodeosVersion, err error) {
+	matches := nodeosVersionRegexp.FindAllStringSubmatch(version, -1)
+	if len(matches) == 0 {
+		err = fmt.Errorf("unable to parse version %q, expected to match %s", version, nodeosVersionRegexp)
+		return
+	}
+
+	userLog.Debug("nodeos version regexp matched", zap.Reflect("matches", matches))
+
+	// We don't care for multiple matches for now
+	match := matches[0]
+	out.full = match[0]
+
+	// We skip the errors since the regex match only digits on those groups
+	out.major, _ = strconv.Atoi(match[1])
+	out.minor, _ = strconv.Atoi(match[2])
+	out.patch, _ = strconv.Atoi(match[3])
+
+	if len(match) >= 5 {
+		out.suffix = match[5]
+	}
+
+	return
+}
+
+func (v nodeosVersion) String() string {
+	return v.full
+}
+
+func (v nodeosVersion) supportsDeepMind(deepMindMajor int) bool {
+	// FIXME: This check is good only for releases prepared by dfuse Team directly.
+	//        When we are going to use the upstream version of EOSIO, this is not going
+	//        to work as expected! At the same time, there is nothing else that can be
+	//        done just now, we could check that he version is above a certain value.
+	if !strings.Contains(v.suffix, "dm") {
+		return false
+	}
+
+	matches := deepMindVersionRegexp.FindAllStringSubmatch(v.suffix, -1)
+	if len(matches) != 1 {
+		userLog.Debug("unable to parse deep mind version", zap.String("deep_mind_version", v.suffix), zap.Stringer("regexp", deepMindVersionRegexp))
+		return false
+	}
+
+	match := matches[0]
+
+	// We skip the errors since the regex match only digits on those groups
+	major, _ := strconv.Atoi(match[1])
+
+	return major == deepMindMajor
+}
+
+func maybeCheckNodeosVersion() {
+	if viper.GetBool("skip-checks") {
+		return
+	}
+
+	version, err := newNodeosVersionFromSystem()
+	if err != nil {
+		userLog.Debug("unable to extract nodeos version from system", zap.Error(err))
+		cliErrorAndExit(dedentf(`
+			We were unable to detect "nodeos" version on your system. This can be due to
+			one of the following reasons:
+			- You don't have "nodeos" installed on your system
+			- It's installed by no referred by your PATH environment variable, so we did not find it
+			- It's installed but execution of "nodeos --version" failed
+
+			Make sure you have a dfuse instrumented 'nodeos' binary, follow instructions
+			at https://github.com/dfuse-io/dfuse-eosio/blob/develop/DEPENDENCIES.md#dfuse-instrumented-eosio-prebuilt-binaries
+			to find how to install it.
+
+			If you have your dfuse instrumented 'nodeos' binary outside your PATH, use --nodeos-path=<location>
+			argument to specify path to it.
+
+			If you think this is a mistake, you can re-run this command adding --skip-checks, which
```

Comment:
si on change le flag, changer cette ligne aussi ^^

Reply:
Yeah sure, I'll check `develop` branch to be sure it's aligned, let's hope otherwise the release is a little bit "screwed".